app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,13 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.posts
+  end
 
-  def show; end
+  def show
+    logger.debug(params)
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:id])
+    @comments = @post.recent_comments
+  end
 end
```

Comment:
- [ **OPTIONAL** ] **RECOMMENDED:** I noticed that you made a mistake while trying to fix the previous changes. I think it would be better if you update the above `@comments = @post.recent_comments` to this `@comments = @user.comments` as in your `posts#show` page it displays five recent comment of a Post instead of displaying all the comment of a post. `Recent Comments` should be rendered on the `post#index` page.